replit_river/codegen/client.py

```diff
@@ -623,24 +635,20 @@ def __init__(self, client: river.Client[{handshake_type}]):
                 parse_output_method = "lambda x: None"
 
             if procedure.type == "rpc":
-                control_flow_keyword = "return "
-                if output_type == "None":
-                    control_flow_keyword = ""
-
                 current_chunks.extend(
                     [
                         reindent(
                             "  ",
                             f"""\
                     async def {name}(
                       self,
-                      input: {input_type},
+                      init: {init_type},
                     ) -> {output_type}:
-                      {control_flow_keyword}await self.client.send_rpc(
-                        '{schema_name}',
-                        '{name}',
-                        input,
-                        {render_input_method},
+                      return await self.client.send_rpc(
```

Comment:
all this stuff is hard to review (and I'm sure iterate on), we should set up some simple tests. Snapshot testing is probably good enough https://pypi.org/project/pytest-snapshot/.

I did this for river js https://github.com/replit/river/pull/261 using https://vitest.dev/guide/snapshot

Reply:
The best way I have found to iterate on this is to just run the codegen against different production schemas and verify that the set of changes is what I expect. I'm happy to wait until after I land the babel changes before merging this.

The codegen currently isn't compositional enough to make snapshot testing manageable. The more I work with it though I've been trying to come up with strategies to make it less imperative without increasing cognitive load too much, I'll land snapshot testing then.